### PR TITLE
fix a bug in windows

### PR DESCRIPTION
**Pull request by GZShi**
The `SOCKET` type is defined as a unsigned number in windows, the `evutil_socket_t` variables are always large than zero, therefore the condition not work well here.

### CODE REVIEW

File: lib/cpp/src/thrift/server/TNonblockingServer.cpp [hunk 1]
```diff
@@ -1324,7 +1324,7 @@ TNonblockingIOThread::~TNonblockingIOThread() {
     ownEventBase_ = false;
   }
 
-  if (listenSocket_ >= 0) {
+  if (listenSocket_ >= 0 && listenSocket_ != THRIFT_INVALID_SOCKET) {
```
Comment (jeking3) [inline lib/cpp/src/thrift/server/TNonblockingServer.cpp]: Perhaps the only test needed is to see if the socket is not an invalid socket?  Is the >= 0 test necessary at this point?

Comment (Jens-G) [inline lib/cpp/src/thrift/server/TNonblockingServer.cpp]: @GZShi: Any comments from your side?

Comment (GZShi) [inline lib/cpp/src/thrift/server/TNonblockingServer.cpp]: I agree with @jeking3 .
@Jens-G thanks for reply :)

Comment (Jens-G) [inline lib/cpp/src/thrift/server/TNonblockingServer.cpp]: @GZShi : would you mind updating your PR accordingly?

**Comment by Jens-G** [comment]
Tracked in THRIFT-4038
@GZShi: next time please have a look at our [contrib guidelines](http://thrift.apache.org/docs/HowToContribute)

**Pull request closed without merging**